xclim/core/indicator.py

```diff
@@ -1355,11 +1355,11 @@ def _show_deprecation_warning(self):
         )
 
 
-class ResamplingIndicator(Indicator):
-    """Indicator that performs a resampling computation.
+class CheckMissingIndicator(Indicator):
+    """Class for indicators that completely reduce the time dimension, adding missing value checks.
 
-    Compared to the base Indicator, this adds the handling of missing data,
-    and the check of allowed periods.
+    A full reduction of the "time" dimension is expected by default: the missing step will fail if the output still has a time dimension.
+    To enable resampling, the :py:meth:`_get_missing_freq` method can be subclassed to return the resampling frequency. The method is always called with the indicator parameters.
```

Comment:
Now that I've read the code I understand this comment, but it was cryptic on the first pass. 

It feels semantically weird that CheckMissingIndicator would be defined as an Indicator reducing the time dimension, but that ResamplingIndicator would then subclass it and revert that property. 

Should we bake missing values checks directly inside `Indicator` now that we're able to support them everywhere?